packages/react-ui/src/app/builder/pieces-selector/create-todo-dialog.tsx

```diff
@@ -0,0 +1,441 @@
+                    <p className="text-sm text-muted-foreground">
+                      {t(
+                        'Users will manage and respond to todos directly within the Activepieces interface. Ideal for internal teams.',
+                      )}
+                    </p>
+                  </div>
+                  <div
+                    className={`p-4 rounded-lg border cursor-pointer transition-all ${
+                      todoType === TodoType.EXTERNAL
+                        ? 'border-primary bg-primary/5'
+                        : 'border-border hover:border-primary/50 hover:bg-muted/50'
+                    }`}
+                    onClick={() => setTodoType(TodoType.EXTERNAL)}
+                    onMouseEnter={() => setHoveredOption(TodoType.EXTERNAL)}
+                    onMouseLeave={() => setHoveredOption(null)}
+                  >
+                    <div className="flex justify-between items-center mb-2">
+                      <h4 className="text-md font-medium">
+                        {t('External Channel (Slack, Teams, Email, ...)')}
+                      </h4>
+                      <div className="flex-shrink-0 w-5 h-5">
+                        <div
+                          className={`w-5 h-5 rounded-full grid place-items-center border ${
+                            todoType === TodoType.EXTERNAL
+                              ? 'border-primary'
+                              : 'border-muted-foreground'
+                          }`}
+                        >
+                          {todoType === TodoType.EXTERNAL && (
+                            <div className="w-3 h-3 rounded-full bg-primary"></div>
+                          )}
+                        </div>
+                      </div>
+                    </div>
+                    <p className="text-sm text-muted-foreground">
+                      {t(
+                        'Send notifications with approval links via external channels like Slack, Teams or Email. Best for collaborating with external stakeholders.',
+                      )}
+                    </p>
+                  </div>
+                </div>
+              </div>
+
+              <div className="md:w-1/2 flex flex-col items-center justify-center">
+                <div className="border rounded-lg overflow-hidden p-3 w-full h-full">
+                  <div className="flex flex-col items-center h-[480px]">
+                    <h3 className="text-md font-medium mb-3 text-center">
+                      {displayImageType === TodoType.INTERNAL
+                        ? t('Preview (Activepieces Todos)')
+                        : t('Preview (External channel)')}
+                    </h3>
+
+                    <div className="w-full h-[350px] overflow-hidden rounded mb-2 flex items-center justify-center bg-muted/50 relative">
+                      <img
+                        src={
+                          displayImageType === TodoType.INTERNAL
```

Comment:
Instead of having if statements everywhere, shouldn't we  have an object or so and just refer it there, so all information related the variance are in one place.


One way you can do it is to make these as paramters, and have another components and call this component based on the conidtion and pass the correct customization there.